tests/xmltester/tests/polygonizeexcludeholes.xml

```diff
@@ -0,0 +1,74 @@
+<run>
```

Comment:
Might be nice to have some self-touching multipolygons as well.  Here's one:

GEOMETRYCOLLECTION (LINESTRING (100 100, 100 400, 400 400, 400 100, 100 100), LINESTRING (150 350, 350 350, 350 150, 100 100, 150 350), LINESTRING (200 250, 250 200, 100 100, 200 250), LINESTRING (200 250, 160 290, 150 250, 200 250), LINESTRING (350 350, 200 300, 200 250, 300 250, 350 350))

with result:

MULTIPOLYGON (((200 250, 250 200, 100 100, 200 250)), ((100 100, 100 400, 400 400, 400 100, 100 100), (150 350, 100 100, 350 150, 350 350, 150 350)), ((200 250, 150 250, 160 290, 200 250)), ((350 350, 300 250, 200 250, 200 300, 350 350)))

Reply:
And here's another complex checkerboard shape:

`MULTILINESTRING ((50 300, 100 300), (50 300, 50 450, 200 450), (300 50, 300 0, 0 0, 0 300, 50 300), (100 150, 50 150, 50 300), (150 150, 100 150), (100 150, 100 300), (150 100, 100 100, 100 150), (100 300, 150 300), (100 300, 100 400, 200 400), (300 100, 150 100), (150 100, 150 150), (300 50, 150 50, 150 100), (300 150, 150 150), (150 150, 150 300), (150 300, 200 300), (150 300, 150 350, 200 350), (200 300, 300 300, 300 200), (200 300, 200 350), (300 200, 200 200, 200 300), (200 350, 350 350), (200 350, 200 400), (200 400, 350 400), (200 400, 200 450), (200 450, 350 450, 350 400), (200 450, 200 500, 500 500, 500 200, 450 200), (450 200, 450 50, 300 50), (300 100, 300 50), (400 200, 400 100, 300 100), (300 150, 300 100), (350 200, 350 150, 300 150), (300 200, 300 150), (350 200, 300 200), (400 200, 350 200), (350 350, 350 200), (350 350, 400 350), (350 400, 350 350), (350 400, 400 400, 400 350), (450 200, 400 200), (400 350, 400 200), (400 350, 450 350, 450 200))`

Result

`MULTIPOLYGON (((150 100, 150 50, 300 50, 300 0, 0 0, 0 300, 50 300, 50 150, 100 150, 100 100, 150 100)), ((200 400, 100 400, 100 300, 50 300, 50 450, 200 450, 200 400)), ((100 150, 100 300, 150 300, 150 150, 100 150)), ((150 150, 300 150, 300 100, 150 100, 150 150)), ((200 350, 200 300, 150 300, 150 350, 200 350)), ((300 200, 200 200, 200 300, 300 300, 300 200)), ((350 400, 350 450, 200 450, 200 500, 500 500, 500 200, 450 200, 450 350, 400 350, 400 400, 350 400)), ((200 350, 200 400, 350 400, 350 350, 200 350)), ((450 200, 450 50, 300 50, 300 100, 400 100, 400 200, 450 200)), ((300 150, 300 200, 350 200, 350 150, 300 150)), ((400 350, 400 200, 350 200, 350 350, 400 350)))`

Reply:
@dr-jts I tried the self-touching polygon and it gives the result you mention.But with the checkerboard shape, I just get a polygon ```POLYGON ((0.00 0.00, 0.00 300.00, 50.00 300.00, 50.00 450.00, 200.00 450.00, 200.00 500.00, 500.00 500.00, 500.00 200.00, 450.00 200.00, 450.00 50.00, 300.00 50.00, 300.00 0.00, 0.00 0.00))``` that is the outermost outline of the checkerboard shape.  PostGIS ST_BuildArea() also return the same (logical given that the GEOS code is a port of liblwgeom/librttopo)

Reply:
Oh well, I guess that's another reason to port the JTS code (it works there... ) . :)

For now either leave the test out or comment it out with a TBD note...

Reply:
Done